### PR TITLE
Appointment activity improvements

### LINKED ISSUE #26: Fix appointment interface and modify BasicAppointment accordingly
Remove the unnecessary setters from the interface and remove the @Overwrite tags in BasicAppointment

### LINKED ISSUE #28: Create private/offline appointment classes / decorators (remains to be seen)

### LINKED ISSUE #29: Change the way that the appointment activity receives the user when created

### LINKED ISSUE #33: Add a settings button to the AppointmentActivity that shows a new window to select the specific settings of the room

### PR DESCRIPTION
**Pull request by Morpheo1**
Added a settings button to further configure an appointment when creating it. Resolves : #26 Resolves : #28 Resolves : #29 Resolves : #33

### CODE REVIEW

**Review by MRandl — APPROVED** [review]
all good, we'll need to rework the user interface to accommodate all the changes but the should not be that much of a burden

File: app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java [hunk 1]
```diff
@@ -72,7 +79,8 @@ protected void onCreate(Bundle savedInstanceState) {
         txtError.setVisibility(View.INVISIBLE);
 
         //We need to know who is trying to create an appointment as they are the owner, see BasicAppointment implementation
-        user = (User) getIntent().getSerializableExtra(EXTRA_USER);
+        user = new DatabaseUser("owner");
```
Comment (MrZaiko) [inline app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java:82]: Use MainUserSingleton() to get the owner id

File: app/src/main/java/io/github/polysmee/appointments/BasicAppointment.java [hunk 2]
```diff
@@ -15,8 +15,18 @@
...
+        this(startTime, duration, course, title, owner, false, new HashSet<>(), new HashSet<>());
+    }
+
+    public BasicAppointment(long startTime, long duration, String course, String title, User owner, boolean isPrivate) {
+        this(startTime, duration, course, title, owner, isPrivate, new HashSet<>(), new HashSet<>());
+    }
+
+    public BasicAppointment(long startTime, long duration, String course, String title, User owner, boolean isPrivate, Set<User> bans, Set<User> invites) {
         if(startTime < 0) {
             this.startTime = 0;
```
Comment (MrZaiko) [inline app/src/main/java/io/github/polysmee/appointments/BasicAppointment.java:31]: Javadoc

**Review by MrZaiko — APPROVED** [review]
Seems good

**Pull request merged**